internal/usecase/repository/domain_repository.go

```diff
@@ -407,50 +420,68 @@ func (r *domainRepository) createIpaDomain(
 	return nil
 }
 
-func (r *domainRepository) updateIpaDomain(
-	db *gorm.DB,
-	data *model.Ipa,
-) (err error) {
-	if db == nil {
-		return internal_errors.NilArgError("db")
-	}
-	if data == nil {
-		return internal_errors.NilArgError("data' of type '*model.Ipa")
-	}
-	if err = db.Unscoped().
-		Delete(data).Error; err != nil {
-		return err
+func (r *domainRepository) ipaFillUpdateAgentReminders(oldData *model.Domain, newData *model.Domain) (remCerts []uint, remServers []uint, remLocations []uint) {
+	remCerts = nil
+	remServers = nil
+	remLocations = nil
+
+	// Check cert reminders
+	OldLenCerts := len(oldData.IpaDomain.CaCerts)
+	NewLenCerts := len(newData.IpaDomain.CaCerts)
+	if OldLenCerts > NewLenCerts {
+		remCerts = make([]uint, OldLenCerts-NewLenCerts)
+		for i := NewLenCerts; i < OldLenCerts; i++ {
+			remCerts[i-NewLenCerts] = oldData.IpaDomain.CaCerts[i].Model.ID
+		}
```

Comment:
I'm not sure whether I understand the algorithm to perform upserts. Does a new cert + removal of an existing cert recycle a database entry? That sounds wrong to me.

Would it help if `ipa-hcc` would include a persistent, unique id that identifies an entry (server, location, cert)? That would allow us match LDAP entries to database entries.

Reply:
About "Does a new cert + removal of an existing cert recycle a database entry?", yes that would be what is doing with this change; the data would be modified, but the internal database key is kept; below what I have observed:

----

- If I add Cert A + remove Cert B,
- the record that store Cert A information after the operation will hold the information of Cert B and the key of the record is kept.
- The DB has no entities that point out to Certs (the certs has a foreign key to the Ipa record hat own them).

> If the DB schema is extended in the future and some entity reference to the cert, then this operation as it is implemented on this PR would be wrong.

----

About "Would it help if ipa-hcc would include a persistent, unique id that identifies an entry (server, location, cert)?", I am not sure; if we update (without upserts) the data, we could split the record to update, the records to insert and the records to delete; for domains with a lot of information we could group on bulk data.

----

Would we prefer then the navy implementation? (I am not against that, as I think this could be an early optimization that maybe we want to delay for later).